### PR TITLE
README should only contain a pitch

### PR DESCRIPTION
**Pull request by breerly**
This reduces the README to a pitch only.

From here, let's expand the Godoc to contain the actual documentation.

At the most, we can circle back to the README to provide a Quickstart in another PR.

Note that the icon is just a placeholder till we get the official branding in.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,121 +1,26 @@
...
-## Container
-
-`package dig` exposes `type Container` as an object capable of resolving a
-directional dependency graph.
-
-To create one:
-
-```go
-import "go.uber.org/dig"
+* Powering an application framework, e.g. dig powers [Fx](https://github.com/uber-go/fx).
```
Comment (glibsm) [inline README.md]: just `e.g. Fx`

File: README.md [hunk 2]
```diff
@@ -1,121 +1,26 @@
-# dig: Dependency Injection Framework for Go
+# :hammer: dig [![GoDoc][doc-img]][doc] [![Build Status][ci-img]][ci] [![Coverage Status][cov-img]][cov] [![Report Card][report-card-img]][report-card]
 
-[![GoDoc][doc-img]][doc]
-[![Coverage Status][cov-img]][cov]
-[![Build Status][ci-img]][ci]
-[![Report Card][report-card-img]][report-card]
+A dependency injection toolkit for Go.
```
Comment (glibsm) [inline README.md]: "reflection based" might be good to shove somewhere in there to hint at the "bad for" section you indicated

File: README.md [hunk 3]
```diff
@@ -1,121 +1,26 @@
...
-**All objects in the container are treated as a singletons**, meaning there can be
-only one object in the graph of a given type.
-
-There are plans to expand the API to support multiple objects of the same type
-in the container, but for time being consider using a factory pattern.
+### Bad for:
 
-## Provide
+* Using in place of an application framework, like [Fx](https://github.com/uber-go/fx).
+* Resolving deps once the process has already started.
```
Comment (glibsm) [inline README.md]: expand word deps

File: README.md (hunk 2 above)
Comment (breerly) [inline README.md]: Agreed - will update!

File: README.md [hunk 4]
```diff
@@ -1,121 +1,26 @@
-# dig: Dependency Injection Framework for Go
+# :hammer: dig [![GoDoc][doc-img]][doc] [![Build Status][ci-img]][ci] [![Coverage Status][cov-img]][cov] [![Report Card][report-card-img]][report-card]
```
Comment (breerly) [inline README.md]: Obviously would love to switch the hammer out for some type of hand shovel:

https://www.iconfinder.com/search/?q=shovel

For the time being, the hammer works well with the idea that dig is a toolkit, not an application framework to be used directly.

**Pull request merged**